zebra-network/src/peer_set/initialize.rs

```diff
@@ -182,47 +188,149 @@ where
     );
 
     handle_tx
-        .send(vec![add_guard, listen_guard, crawl_guard])
+        .send(vec![initial_peers_guard, listen_guard, crawl_guard])
         .unwrap();
 
     (peer_set, address_book)
 }
 
-/// Use the provided `handshaker` to connect to `initial_peers`, then send
-/// the results over `tx`.
-#[instrument(skip(initial_peers, outbound_connector, tx))]
-async fn add_initial_peers<S>(
+/// Use the provided `outbound_connector` to connect to `initial_peers`, then
+/// send the results over `peerset_tx`.
+///
+/// Stop trying peers once we've had `peerset_initial_target_size` successful
+/// handshakes.
+#[instrument(skip(initial_peers, outbound_connector, peerset_tx))]
+async fn add_initial_peers<C>(
     initial_peers: std::collections::HashSet<SocketAddr>,
-    outbound_connector: S,
-    mut tx: mpsc::Sender<PeerChange>,
+    outbound_connector: C,
+    mut peerset_tx: mpsc::Sender<PeerChange>,
+    peerset_initial_target_size: usize,
 ) -> Result<(), BoxError>
 where
-    S: Service<SocketAddr, Response = Change<SocketAddr, peer::Client>, Error = BoxError> + Clone,
-    S::Future: Send + 'static,
+    C: Service<SocketAddr, Response = Change<SocketAddr, peer::Client>, Error = BoxError>
+        + Clone
+        + Send
+        + 'static,
+    C::Future: Send + 'static,
 {
-    info!(?initial_peers, "connecting to initial peer set");
-    // ## Correctness:
+    let initial_peers_len = initial_peers.len();
+    info!(
+        ?initial_peers_len,
+        ?initial_peers,
+        ?peerset_initial_target_size,
+        "connecting to initial peer set"
+    );
+
+    let initial_meta_addr = initial_peers.into_iter().map(MetaAddr::new_seed_meta_addr);
```

Comment:
```suggestion
    let initial_meta_addrs = initial_peers.into_iter().map(MetaAddr::new_seed_meta_addr);
```

Not sure if there's a reason to not make it plural :thinking:

Reply:
Bad habits from database field naming standards?

This code is fixed or deleted in the latest #2160.